docs/azure/cluster-parameters.md

```diff
@@ -0,0 +1,94 @@
+# Azure cluster parameters
+
+## Prerequisites
+
+- Azure CLI installed
+- `az login` command executed
+
+## Cluster Identity
+
+To provide credentials for CAPI Azure provider (CAPZ) the `AzureClusterIdentity`
+resource must be created. This should be done before provisioning any clusters.
+
+
+To create the `AzureClusterIdentity` you should first get the desired
+`SubscriptionID` by executing `az account list -o table` which will return list
+of subscriptions available to user.
+
+Then you need to create service principal which will be used by CAPZ to interact
+with Azure API. To do so you need to execute the following command:
+
+```bash
+az ad sp create-for-rbac --role contributor --scopes="/subscriptions/<Subscription ID>"
+```
+
+The command will return json with the credentials for the service principal which
+will look like this:
+
+```json
+{
+	"appId": "29a3a125-7848-4ce6-9be9-a4b3eecca0ff",
+	"displayName": "azure-cli",
+	"password": "u_RANDOMHASH",
+	"tenant": "2f10bc28-959b-481f-b094-eb043a87570a",
+}
+```
+
+*Note: make sure to save this credentials and treat them like passwords.*
+
+With the data from the json you can now create the `AzureClusterIdentity` object
+and it's secret.
+
+The objects created with the data above can look something like this:
+
+**Secret**:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: az-cluster-identity-secret
+stringData:
+  clientSecret: u_RANDOMHASH
+type: Opaque
+```
+
+**AzureClusterIdentity**:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureClusterIdentity
+metadata:
+  labels:
+    clusterctl.cluster.x-k8s.io/move-hierarchy: "true"
+  name: az-cluster-identity
+spec:
+  allowedNamespaces: {}
+  clientID: 29a3a125-7848-4ce6-9be9-a4b3eecca0ff
+  clientSecret:
+    name: az-cluster-identity-secret
```

Comment:
Should the `namespace` be also provided here?
```
E0902 08:41:10.716796       1 controller.go:329] "Reconciler error" err="failed to create scope: failed to configure azure settings and credentials for Identity: Unable to fetch ClientSecret: an empty namespace may not be set when a resource name is provided" controller="azurecluster" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="AzureCluster" AzureCluster="default/ekaz-test" namespace="default" name="ekaz-test" reconcileID="366dcd31-e606-4360-92bc-58b39923d4bd"
```

Reply:
Thanks, but I meant the namespace of the secret: `clientSecret.namespace`. Without it, the CAPZ controller is failing with an error:
```
E0902 08:41:10.716796       1 controller.go:329] "Reconciler error" err="failed to create scope: failed to configure azure settings and credentials for Identity: Unable to fetch ClientSecret: an empty namespace may not be set when a resource name is provided" controller="azurecluster" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="AzureCluster" AzureCluster="default/ekaz-test" namespace="default" name="ekaz-test" reconcileID="366dcd31-e606-4360-92bc-58b39923d4bd"
```

Reply:
Sorry, misunderstood you first time. Added namespaces explicitly everywhere.